lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@zeroedin-bill and just FYI -- phpinfo() outputs the same thing as PGSQL_LIBPQ_VERSION

8.4.20

YET, client_encoding works fine for me. @riccardonar is saying it doesn't work 9.0 client but I haven't confirmed that. I am surprised it works with 8.4 but not 9.0 though.